sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py

```diff
@@ -0,0 +1,835 @@
+    :keyword azure.webpubsub.client.WebPubSubProtocolType protocol_type: Subprotocol type
+    :keyword int reconnect_retry_total: total number of retries to allow for reconnect. If 0, it means disable
+     reconnect. Default is 3.
+    :keyword float reconnect_retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay). In fixed mode, retry policy will always
+     sleep for {backoff factor}. In 'exponential' mode, retry policy will sleep for:
+     "{backoff factor} * (2 ** ({number of retries} - 1))" seconds. If the backoff_factor is 0.1, then the
+     retry will sleep for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 1.0.
+    :keyword float reconnect_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode reconnect_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword int message_retry_total: total number of retries to allow for sending message. Default is 3.
+    :keyword float message_retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay). In fixed mode, retry policy will always
+     sleep for {backoff factor}. In 'exponential' mode, retry policy will sleep for:
+     "{backoff factor} * (2 ** ({number of retries} - 1))" seconds. If the backoff_factor is 0.1, then the
+     retry will sleep for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 1.0.
+    :keyword float message_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode message_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword bool auto_rejoin_groups: auto_rejoin_groups, default is True
+    :keyword str user_agent: The user agent to be used for the request. If specified, this will be added in front of
+     the default user agent string.
+    :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+    :keyword float ack_timeout: Time limit to wait for ack message from server. The default value is 30.0 seconds.
+    :keyword float start_timeout: Time limit to wait for successful client start. The default value is 30.0 seconds.
+    """
+
+    def __init__(
+        self,
+        credential: Union[WebPubSubClientCredential, str],
+        **kwargs: Any,
+    ) -> None:
+        if isinstance(credential, WebPubSubClientCredential):
+            self._credential = credential
+        elif isinstance(credential, str):
+            self._credential = WebPubSubClientCredential(credential)
+        else:
+            raise TypeError("type of credential must be str or WebPubSubClientCredential")
+        reconnect_retry_total = kwargs.pop("reconnect_retry_total", _RETRY_TOTAL)
```

Comment:
are these defaults synced to the defaults in azure core? we should ideally rely on the defaults in azure core instead of duplicating that in case anything changes

Reply:
retry policy in azure-core is for http and this package is for websocket so we can't use default retry policy directly. But I use similar name to try to keep consistency with [azure-core](https://github.com/Azure/azure-sdk-for-python/blob/f368f19b52229bbc29d50929b39bcb12e45c4b5c/sdk/core/azure-core/azure/core/pipeline/policies/_retry.py#L57-L72)